pytest_parallel/__init__.py

```diff
@@ -62,12 +62,41 @@ def process_with_threads(config, queue, session, tests_per_worker, errors):
     # so we know we are running as a worker.
     config.parallel_worker = True
 
-    threads = []
-    for _ in range(tests_per_worker):
-        thread = ThreadWorker(queue, session, errors)
-        thread.start()
-        threads.append(thread)
-    [t.join() for t in threads]
+    if tests_per_worker == 1:
+        worker_run(queue, session)
+    else:
+        threads = []
+        for _ in range(tests_per_worker):
+            thread = ThreadWorker(queue, session, errors)
+            thread.start()
+            threads.append(thread)
+        [t.join() for t in threads]
+
+
+def worker_run(queue, session):
+    pickling_support.install()
+    while True:
+        try:
+            index = queue.get()
+            if index == 'stop':
+                queue.task_done()
+                break
+        except ConnectionRefusedError:
+            time.sleep(.1)
+            continue
+        item = session.items[index]
+        try:
+            run_test(session, item, None)
+        except BaseException:
+            import pickle
+            import sys
+
+            self.errors.put(('name', pickle.dumps(sys.exc_info())))
```

Comment:
We seem to have lost a `name` variable here - should that be passed in?

Reply:
Oh right. Done